wormhole-connect/src/hooks/useRoutesQuotesBulk.ts

```diff
@@ -60,16 +70,34 @@ const useRoutesQuotesBulk = (routes: string[], params: Params): HookReturn => {
     // Forcing TS to infer that fields are non-optional
     const rParams = params as Required<QuoteParams>;
 
-    setIsFetching(true);
-    config.routes.getQuotes(routes, rParams).then((quoteResults) => {
-      if (!unmounted) {
-        setQuotes(quoteResults);
-        setIsFetching(false);
-      }
-    });
+    const onComplete = () => {
+      // Refresh quotes in 20 seconds
+      const refreshTimeout = setTimeout(
+        () => setNonce(new Date().valueOf()),
+        QUOTE_REFRESH_INTERVAL,
+      );
+      setRefreshTimeout(refreshTimeout);
+    };
+
+    if (isTransactionInProgress) {
+      // Don't fetch new quotes if the user has committed to one and has initiated a transaction
+      onComplete();
```

Comment:
Should we cancel the `refreshTimeout` here?

Reply:
no - all we're doing is queueing up the next iteration in 20 seconds.